### ISSUE TITLE
Permissible states not respecting guard parameters

### ISSUE THREAD

**Issue by froske** [aasm/aasm#388]
I understand calling for permissible states works with guards (ref issue #54), but I can't get it to work when my guard has a parameter.  It seems like it's not getting passed to the guard (only when I call for permissible states.

Example:

```
state :new, :reviewed, :finalized

event :mark_as_reviewed do
  transitions :from => :new, :to => :reviewed, :guards => [:user_is_manager?]
end

...

def user_is_manager?(usr)
  ok = false
  if usr.has_role? :manager
    ok = true
  end
  return ok
end
```

```
$ u = User.find(100)
$ u.has_role? :manager
=> true

$item.aasm.current_state
=> :new

$ item.may_mark_as_reviewed?(u)
=> true

$ item.aasm.states(:permitted => true, :usr => u).map(&:name)
=> NoMethodError: undefined method `has_role?' for nil:NilClass.....

```

but the guard works in other instances with a parameter, like when I actually try to change the state.

```
$ item.mark_as_reviewed!(u)
=> true

$ item.aasm.current_state
=> :reviewed
```

If I do a `puts usr.inspect` inside the guard method, it shows me that the User object did in fact make it when I called `item.may_mark_reviewed?(u)`  but when I call for permitted states it's just nil.

I've tried every syntax I can think of. 

What am I missing?

**Comment by anilmaurya** [comment]
Fixed with #389

**Comment by anilmaurya** [comment]
@froske Thank you for reporting this bug, its fixed now in version `4.11.1 `.
